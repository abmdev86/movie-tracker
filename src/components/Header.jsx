import {
    AppBar,
    Box,
    Container,
    IconButton,
    Menu,
    MenuItem,
    Toolbar,
    Typography,
    useTheme,
} from "@mui/material";
import MenuIcon from "@mui/icons-material/Menu";
import PropTypes from "prop-types";
import { useState } from "react";
import { Link } from "react-router-dom";

export default function Header({ pages, children }) {
    const [anchorElNav, setAnchorElNav] = useState(null);
    const theme = useTheme();
    const handleOpenNavMenu = (event) => {
        setAnchorElNav(event.currentTarget);
    };

    const handleCloseNavMenu = () => {
        setAnchorElNav(null);
    };
    return (
        <AppBar position="static" color="secondary" sx={{ width: { xl: "100%" } }} enableColorOnDark>
            <Container maxWidth="xl">
                <Toolbar variant="dense" disableGutters>
                    <Link
                        to="/"
                        style={{
                            letterSpacing: ".3rem",
                            color: "inherit",
                            textDecoration: "none",
                        }}
                    >
                        <Typography
                            sx={{
                                marginRight: 2,
                                display: { xs: "none", md: "flex" },
                                fontFamily: "monospace",
                                fontWeight: 700,
                                letterSpacing: ".3rem",
                                color: "inherit",
                                textDecoration: "none",
                            }}
                        >
                            Movie Tracker
                        </Typography>
                    </Link>

                    <Box sx={{ flexGrow: 1, display: { xs: "flex", md: "none" }, bgcolor: `${theme.palette.secondary.main}` }}>
                        <IconButton
                            size="large"
                            aria-label="account of current user"
                            aria-controls="menu-appbar"
                            aria-haspopup="true"
                            onClick={handleOpenNavMenu}
                            color="inherit"
                        >
                            <MenuIcon />
                        </IconButton>
                        <Menu
                            id="menu-appbar"
                            anchorEl={anchorElNav}
                            anchorOrigin={{
                                vertical: "bottom",
                                horizontal: "left",
                            }}
                            keepMounted
                            transformOrigin={{
                                vertical: "top",
                                horizontal: "left",
                            }}
                            open={Boolean(anchorElNav)}
                            onClose={handleCloseNavMenu}
                            sx={{
                                display: { xs: "block", md: "none" },

                            }}
                        >
                            {pages.map((page, index) => (
                                <MenuItem key={index} onClick={handleCloseNavMenu}>
                                    <Link to={`/${page.toLocaleLowerCase()}`}>
                                        <Typography noWrap sx={{ textDecoration: "none" }}>
                                            {page}
                                        </Typography>
                                    </Link>
                                </MenuItem>
                            ))}
                        </Menu>
                    </Box>

                    {/* add icon */}
                    <Link
                        to="/"
                        style={{
                            marginRight: 2,
                            flexGrow: 1,
                            fontFamily: "monospace",
                            fontWeight: 700,
                            letterSpacing: ".3rem",
                            color: "inherit",
                            textDecoration: "none",
                        }}
                    >
                        <Typography
                            variant="h5"
                            noWrap
                            sx={{
                                mr: 2,
                                display: { xs: "flex", md: "none" },
                            }}
                        >
                            Movie Tracker
                        </Typography>
                    </Link>

                    <Box sx={{ flexGrow: 1, display: { xs: "none", md: "flex" }, }}>
                        {pages.map((page, index) => (
                            <Link
                                style={{ textDecoration: "none", color: "inherit" }}
                                key={index}
                                to={`/${page.toLocaleLowerCase()}`}
                            >
                                <Typography
                                    key={index}
                                    sx={{
                                        mr: 2,
                                        display: "block",
                                        letterSpacing: ".1rem",
                                    }}
                                >
                                    {page}
                                </Typography>
                            </Link>
                        ))}
                    </Box>
                    {children}
                </Toolbar>
            </Container>
        </AppBar>
    );
}

Header.propTypes = {
    pages: PropTypes.arrayOf(PropTypes.string),
};
