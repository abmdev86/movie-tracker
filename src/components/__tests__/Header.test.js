import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";

import Header from "../Header";

it("component has container and children", () => {
  const headerComponent = renderer.create(<Header />);
  let tree = headerComponent.toJSON();
  expect(tree).toMatchSnapshot();
});

test("it renders Movie Tracker", () => {
  render(<Header />);
  const headerElement = screen.getByText("Movie Tracker");

  expect(headerElement).toBeInTheDocument();
});
