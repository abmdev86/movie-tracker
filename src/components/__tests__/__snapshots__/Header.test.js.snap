// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component has container and children 1`] = `
<div
  className="MuiBox-root css-i9gxme"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <button
        aria-label="open-drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <div
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1futezt-MuiTypography-root"
      >
        Movie Tracker
      </div>
    </div>
  </header>
</div>
`;
